magpie/permissions.py

```diff
@@ -128,13 +154,40 @@ def __ne__(self, other):
     def __lt__(self, other):
         # type: (Any) -> bool
         """
-        Sort by permission name, followed by *more permissive access* and *more generic scope*.
+        Ascending sort of permission according to their name, access and scope modifiers.
+
+        First sort by permission name alphabetically, followed by increasing *restrictive access* and increasing
+        *range of scoped resources*.
 
         Using this sorting methodology, similar permissions by name are grouped together first, and permissions of same
-        name with modifiers are then ordered from ``allow-recursive`` to ``deny-match``, the first having less priority
-        in the :term:`Effective Permissions` resolution than the later. Respecting :attr:`Access.DENY` is more important
-        than :attr:`Access.ALLOW` (to protect the :term:`Resource`), and :attr:`Scope.MATCH` is *closer* to the actual
-        :term:`Resource` than :attr:`Scope.RECURSIVE` permission received from a *farther* parent in the hierarchy.
+        name with modifiers are then ordered, the first having less priority when selecting a single item to display
+        with conflicting possibilities. Respecting :attr:`Access.DENY` is more important than :attr:`Access.ALLOW`
+        (to protect the :term:`Resource`), and :attr:`Scope.MATCH` is *closer* to the actual :term:`Resource` than
+        :attr:`Scope.RECURSIVE` permission received from a *farther* parent in the hierarchy.
+
+        Explicitly, sorting becomes::
+
+            [name1]-[allow]-[match]
+            [name1]-[allow]-[recursive]
+            [name1]-[deny]-[match]
+            [name1]-[deny]-[recursive]
+            [name2]-[allow]-[match]
+            [name2]-[allow]-[recursive]
+            [name2]-[deny]-[match]
+            [name2]-[deny]-[recursive]
+            ...
+
+        We then obtain two **crucial** ordering results:
+            1. We can easily pick the last sorted item with highest resolution priority to find the final result of
+               corresponding permissions.
+               (note: final result for same user or group, their direct/inherited resolution is not considered here).
+            2. Picking the first element with lowest priority also displays the permission that impacts the widest
+               range of resources. For instance in Magpie UI, indicating that a permission as :attr:`Scope.RECURSIVE`
+               is more verbose as it tell other resources under it are also receive the specified :class:`Access`
+               modifier rather than only the punctual resource.
+
+        .. warning::
```

Comment:
I wonder, if the warning should not be duplicated at the top too given how bad a modification could be.

Reply:
I'll move it toward the top so it is the second thing listed after minimal description. Harder to miss. I'll also describe it a bit better.